src/Peering/Peering/PeeringService/Prefix/DeleteAzurePeeringServicePrefixCommand.cs

```
@@ -0,0 +1,165 @@
+        public override void ExecuteCmdlet()
+        {
+            try
+            {
+                this.ConfirmAction(
+                    Force.IsPresent,
+                    string.Format(Resources.ProcessMessage, this.Name),
+                    Resources.ContinueMessage,
+                    this.Name,
+                    () =>
+                    {
+                        if (this.ResourceId != null)
+                        {
+                            var resourceId = new ResourceIdentifier(this.ResourceId);
+                            this.ResourceGroupName = resourceId.ResourceGroupName;
+                            this.Name = resourceId.ResourceName;
+                            this.PeeringServiceName = resourceId.ParentResource.Split('/')?[1];
+                        }
+                        if (this.InputObject != null)
+                        {
+                            var resourceId = new ResourceIdentifier(this.InputObject.Id);
+                            this.ResourceGroupName = resourceId.ResourceGroupName;
+                            this.Name = resourceId.ResourceName;
+                            this.PeeringServiceName = resourceId.ParentResource.Split('/')?[1];
+                        }
+                        try
+                        {
+                            if (this.ShouldProcess(string.Format(Resources.ShouldProcessMessage, "null. Unless the -PassThru is present which it will return a boolean, true for success of false for failure.")))
+                            {
+                                this.PeeringServicePrefixesClient.Delete(this.ResourceGroupName, this.PeeringServiceName, this.Name);
+                                if (this.PassThru.IsPresent)
+                                {
+                                    WriteObject(true);
+                                }
+                            }
+                        }
+                        catch (Exception ex)
```

@rileymckenna I don't think we want to catch this exception, I think we want to throw any exception that's received from the server